### PR TITLE
[Add-to-app] Add Xcode 16 warning to CocoaPods instructions

### PR DESCRIPTION
**Pull request by loic-sharma**
CocoaPods does not work with Xcode 16. Add a warning to affected docs. Once CocoaPods fixes this issue, we can remove this warning.

Part of https://github.com/flutter/flutter/issues/153402

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/_includes/docs/add-to-app/ios-project/embed-cocoapods.md [hunk 1]
```diff
@@ -54,6 +54,15 @@ This section presumes you called your Swift app `MyApp`.
    navigate to the root of your app directory.
    Use the `pod init` command to create the `Podfile` file.
 
+   :::warning
+   CocoaPods 1.15.2 does not support Xcode 16.
```
Comment (loic-sharma) [inline src/_includes/docs/add-to-app/ios-project/embed-cocoapods.md:58]: @sfshaza2, I know this doesn't follow the "timeless documentation" best practice. I wasn't sure how to best phrase this. Let me know if you have suggestions!

Comment (parlough) [reply src/_includes/docs/add-to-app/ios-project/embed-cocoapods.md:58]: I think it's fine if this situation is temporary, perhaps:

```suggestion
   CocoaPods (as of version 1.15.2) does not support Xcode 16.
```

That also accounts for versions of Cocoapods before 1.15.2.

File: src/_includes/docs/add-to-app/ios-project/embed-cocoapods.md [hunk 2]
```diff
@@ -54,6 +54,15 @@ This section presumes you called your Swift app `MyApp`.
...
+   If the `pod init` command errors, consider re-creating your Xcode project
+   using Xcode 15 or older.
+
+   For more information, see
```
Comment (parlough) [inline src/_includes/docs/add-to-app/ios-project/embed-cocoapods.md:62]: Optionally consider providing more specific details before linking elsewhere:

```suggestion
   To learn more about the CocoaPods issue, check out
```

**Review by parlough — APPROVED** [review]
Thanks for thinking to add this useful warning!

**Review by sfshaza2 — APPROVED** [review]
Yes, thank you, @loic-sharma!

**Comment by sfshaza2** [comment]
@parlough, someone did something "interesting" on this PR. A change is somehow not attached, but looks ok to me. Can you merge and land?

**Pull request merged**